File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
	public int solution(int k, int[] tangerine) {
		int sum = 0;
		int cnt = 0;
		HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();

		for (int n : tangerine) {
			map.put(n, map.getOrDefault(n, 0) + 1);
		}
		ArrayList<Integer> valueList = new ArrayList<>(map.values());
        Collections.sort(valueList, Collections.reverseOrder());
		for (int v : valueList) {
			if (sum + v >= k) {
				cnt++;
				break;
			} else {
				sum += v;
				cnt++;
			}
		}

		return cnt;
	}
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.kt
class Solution {
    fun solution(k: Int, tangerine: IntArray): Int {
        var answer: Int = 0
        var k = k

        val listMap = hashMapOf<Int, Int>()
        tangerine.forEach {
            listMap[it] = (listMap[it] ?: 0) + 1
        }
        val valueList = listMap.values.sortedDescending()

        var cnt = 0
        var sum = 0
        while (k > sum) {
            sum += valueList[cnt]
            answer++
            cnt++
        }
        return answer
    }
}